### PR TITLE
[Merged by Bors] - fix: Fix `e₁ ≪≫ₗ e₂`

### PR DESCRIPTION
**Pull request by Komyyy**
---

This PR fixes the invalid notation of `LinearEquiv.trans`, `e₁ ≪≫ₗ e₂`.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Equiv.lean [hunk 1]
```diff
@@ -334,7 +335,7 @@ set_option quotPrecheck false in
 /-- The notation `e₁ ≪≫ₗ e₂` denotes the composition of the linear equivalences `e₁` and `e₂`. -/
 infixl:80 " ≪≫ₗ " =>
   @LinearEquiv.trans _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ (RingHom.id _) (RingHom.id _) (RingHom.id _)
-    (RingHom.id _) (RingHom.id _) (RingHom.id _) RingHomCompTriple.ids RingHomCompTriple.ids
+    (RingHom.id _) (RingHom.id _) (RingHom.id _) RingHomCompTriple.ids
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Equiv.lean:338]: I'm a bit worried by this; why does Lean3 take an extra instance argument here?

Comment (Komyyy) [reply Mathlib/Algebra/Module/Equiv.lean:338]: Should we give `LinearEquiv.trans` an extra instance?

File: Mathlib/Algebra/Module/Equiv.lean [hunk 2]
```diff
@@ -322,9 +322,10 @@ variable {re₃₂ : RingHomInvPair σ₃₂ σ₂₃} [RingHomInvPair σ₃₁
 
 variable (e₁₂ : M₁ ≃ₛₗ[σ₁₂] M₂) (e₂₃ : M₂ ≃ₛₗ[σ₂₃] M₃)
 
--- Note: The linter thinks the `RingHomCompTriple` argument is doubled -- it is not.
+-- Porting note: The linter don't thinks the `RingHomCompTriple` argument is doubled, so
+--               `nolint unusedArguments` is not necessary.
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Equiv.lean:326]: This comment is inaccurate, see #18458

File: Mathlib/Algebra/Module/Equiv.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Module/Equiv.lean:337]: Yes, I think we should; though I can't work out a nice way to do so without repeating all the `variables` lines

Comment (eric-wieser) [reply Mathlib/Algebra/Module/Equiv.lean]: I created [a Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/forcing.20inclusion.20of.20unused.20arguments/near/328480100)

File: Mathlib/Algebra/Module/Equiv.lean [hunk 3]
```diff
@@ -322,10 +322,15 @@ variable {re₃₂ : RingHomInvPair σ₃₂ σ₂₃} [RingHomInvPair σ₃₁
 
 variable (e₁₂ : M₁ ≃ₛₗ[σ₁₂] M₂) (e₂₃ : M₂ ≃ₛₗ[σ₂₃] M₃)
 
--- Note: The linter thinks the `RingHomCompTriple` argument is doubled -- it is not.
+set_option linter.unusedVariables false in
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Equiv.lean:325]:
```suggestion
-- Porting note: Lean 4 aggressively removes unused variables declared using `variables`, so
-- we have to list all the variables explicitly here in order to match the Lean 3 signature.
set_option linter.unusedVariables false in
```

File: Mathlib/Algebra/Module/Equiv.lean [hunk 4]
```diff
@@ -322,10 +322,15 @@ variable {re₃₂ : RingHomInvPair σ₃₂ σ₂₃} [RingHomInvPair σ₃₁
...
 /-- Linear equivalences are transitive. -/
 @[trans, nolint unusedArguments]
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Equiv.lean:327]: Please copy the comment from https://github.com/leanprover-community/mathlib/pull/18458 too

Comment (eric-wieser) [reply Mathlib/Algebra/Module/Equiv.lean:329]: This comment still applies

Comment (Komyyy) [reply Mathlib/Algebra/Module/Equiv.lean:331]: I've mistaken this for the porting note, sorry.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**